### PR TITLE
update security context to drop all capabilities & set seccomp 

### PR DESCRIPTION
**Pull request by akoserwal**
Fix:
Warning: would violate PodSecurity "restricted:v1.24": unrestricted capabilities (container "manager" must set securityContext.capabilities.drop=["ALL"]), seccompProfile (pod or container "manager" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")

* `container "manager" must set securityContext.capabilities.drop=["ALL"]`: Drop all capabilities and add only those required 

* https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-seccomp-profile-for-a-container

### CODE REVIEW

**Comment by guicassolato** [comment]
Hi, @akoserwal. Thanks for this PR!

Do you mind running `make bundle` to also update the bundle manifests, please?

File: bundle/manifests/authorino-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -283,6 +283,12 @@ spec:
                 resources: {}
                 securityContext:
                   allowPrivilegeEscalation: false
+                  capabilities:
+                    drop:
+                    - ALL
```
Comment (guicassolato) [inline bundle/manifests/authorino-operator.clusterserviceversion.yaml:288]: Just wondering... We already run as non-root as escalation of privilege is not allowed. Isn't `capabilities.drop: ["ALL"]` redundant in this case?

```sh
$ docker run --cap-add SYS_ADMIN --user 1001 --entrypoint /bin/grep quay.io/kuadrant/authorino-operator:latest -- Cap /proc/self/status
CapInh:	0000000000000000
CapPrm:	0000000000000000
CapEff:	0000000000000000
CapBnd:	00000000a82425fb
CapAmb:	0000000000000000

$ docker run --cap-add SYS_ADMIN --user 1001 --entrypoint /bin/grep <image-based-on-this-branch> -- Cap /proc/self/status
CapInh:	0000000000000000
CapPrm:	0000000000000000
CapEff:	0000000000000000
CapBnd:	00000000a82425fb
CapAmb:	0000000000000000
```

Comment (akoserwal) [reply bundle/manifests/authorino-operator.clusterserviceversion.yaml:288]: Yes, I agree. when I was deploying the operator, I got this warning

```
Warning: would violate PodSecurity "restricted:v1.24": unrestricted capabilities (container "manager" must set securityContext.capabilities.drop=["ALL"]), seccompProfile (pod or container "manager" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost") 
```
I think "dropping all capabilities" will only reduce any further risk of remote code execution, which seems more like a good-to-have thing. run as non-root already handle it. When I added both recommendations, it suppressed the warning.

Comment (guicassolato) [reply bundle/manifests/authorino-operator.clusterserviceversion.yaml:288]: > When I added both recommendations, it suppressed the warning.

Good enough for me.

Thanks, @akoserwal!

**Review by guicassolato — APPROVED** [review]

**Review by guicassolato — APPROVED** [review]

**Pull request merged**